pkg/binding/definition.go

```diff
@@ -38,42 +37,44 @@ const (
 //go:generate mockgen -destination=mocks/mocks.go -package=mocks . Definition,Value
 
 type Definition interface {
-	GetPath() []string
 	Apply(u *unstructured.Unstructured) (Value, error)
+	GetPath() string
 }
 
 type DefinitionBuilder interface {
 	Build() (Definition, error)
 }
 
-type stringDefinition struct {
-	outputName string
-	path       []string
+type definition struct {
+	path string
 }
 
-var _ Definition = (*stringDefinition)(nil)
+func (d *definition) GetPath() string {
+	return d.path
+}
 
-func (d *stringDefinition) getOutputName() string {
-	outputName := d.outputName
-	if len(outputName) == 0 {
-		outputName = d.path[len(d.path)-1]
-	}
-	return outputName
+type stringDefinition struct {
+	outputName string
+	definition
 }
 
-func (d *stringDefinition) GetPath() []string { return d.path[0 : len(d.path)-1] }
+var _ Definition = (*stringDefinition)(nil)
 
 func (d *stringDefinition) Apply(u *unstructured.Unstructured) (Value, error) {
-	val, ok, err := unstructured.NestedFieldCopy(u.Object, d.path...)
+	if d.outputName == "" {
+		return nil, fmt.Errorf("cannot use generic service.binding annotation for string elements, need to specify binding key like service.binding/foo")
+	}
+	val, err := getValuesByJSONPath(u.Object, d.path)
 	if err != nil {
 		return nil, err
 	}
-	if !ok {
-		return nil, errors.New("not found")
+
+	if len(val) != 1 {
+		return nil, fmt.Errorf("Only one value should be returned for %v but we got %v", d.path, val)
```

Comment:
Error strings should not be capitalized.
Ref. https://github.com/golang/go/wiki/CodeReviewComments#error-strings